### DISCUSSION TITLE
Compile failed

### DISCUSSION THREAD

**Discussion by Dudis2k** [apache/amoro#3321] [Q&A]
Facing error - Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.8.1:compile (scala-compile-first) on project amoro-format-mixed-spark-3.3: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.8.1:compile failed.: CompileFailed

cmd executed: mvn clean package -DskipTests

My System configurations:
MacOs
Jdk8
Scala version 2.12
Spark 3.5.3

**Comment by xxubai** [comment]
Hi @Dudis2k , Sorry for the late reply, please try the following command to compile a separate module, remember to execute it in the root directory of the project:
```
mvn clean package -DskipTests -pl 'amoro-mixed-format/amoro-mixed-format-spark/v3.3/amoro-mixed-format-spark-3.3'
```

**Reply by TheR1sing3un** [reply]
> Hi @Dudis2k , Sorry for the late reply, please try the following command to compile a separate module, remember to execute it in the root directory of the project:
> 
> ```
> mvn clean package -DskipTests -pl 'amoro-mixed-format/amoro-mixed-format-spark/v3.3/amoro-mixed-format-spark-3.3'
> ```

Hi, I meet a same problem like above, but could you please list the entire compilation command in that environment? Looking forward to your reply. And put these commands in `README.md` may help more new users/developers!

**Comment by TheR1sing3un** [comment]
> Facing error - Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.8.1:compile (scala-compile-first) on project amoro-format-mixed-spark-3.3: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.8.1:compile failed.: CompileFailed
> 
> cmd executed: mvn clean package -DskipTests
> 
> My System configurations: MacOs Jdk8 Scala version 2.12 Spark 3.5.3

Are you using zulu-jdk to compile? Please use oracle-jdk, it appears to be a compilation error caused by a difference in the jdk. https://github.com/apache/amoro/pull/3364
